src/main/java/org/fcrepo/spec/testsuite/event/NotificationTest.java

```diff
@@ -0,0 +1,274 @@
+        // PUT emits 1 event
+        await().atMost(TEN_SECONDS).until(() -> listener.stream()
+                .count() == 1);
+
+        listener.clear();
+        doPatch(location, new Headers(new Header("Content-type", "application/sparql-update")),
+                "prefix dc: <http://purl.org/dc/elements/1.1/> INSERT { <> dc:title \"This has been updated\"} " +
+                        "WHERE {}");
+        // PATCH emits one event
+        await().atMost(TEN_SECONDS).until(() -> listener.stream()
+                .count() == 1);
+
+        listener.clear();
+        doDelete(location);
+        // Delete emits 2 events.
+        await().atMost(TEN_SECONDS).until(() -> listener.stream()
+                .count() == 2);
+
+        consumer.close();
+
+        session.close();
+        connection.close();
+    }
+
+    /**
+     * 6.2-A Event Serialization
+     *
+     * @param uri the repository base uri
+     * @throws JMSException problems connecting to broker
+     * @throws InterruptedException interrupt the thread.sleep
+     * @throws UnsupportedEncodingException
+     */
+    @Test(groups = { "MUST" })
+    @Parameters({ "param1" })
+    public void testEventSerialization(final String uri) throws JMSException, InterruptedException,
+            UnsupportedEncodingException {
+        final TestInfo info = setupTest("6.2-A",
+                "The notification serialization MUST conform to the [activitystreams-core] specification, " +
+                        "and each event MUST contain the IRI of the resource and the event type.",
+                "https://fedora.info/2018/06/25/spec/#notification-serialization", ps);
+        // Start with a clean JMS connection.
+        resetConnection();
+        // Get a new consumer.
+        final MessageConsumer consumer = getConsumer();
+        // Assign a message bank to capture the messages.
+        consumer.setMessageListener(new MessageBank());
+        // Start listening to the broker.
+        connection.start();
+        // Do your actions.
+        final Response response = createBasicContainer(uri, info);
+        final String location = getLocation(response);
+        // Get the message bank back.
+        final MessageBank listener = (MessageBank) consumer.getMessageListener();
+        await().atMost(TEN_SECONDS).until(() -> listener.stream()
+                .count() == 2);
+
+        final Resource locResource = ResourceFactory.createResource(location);
+        final List<Message> message = listener.stream().collect(toList());
+        for (final Message m : message) {
+            if (m instanceof TextMessage) {
+                final String body = ((TextMessage) m).getText();
+                final InputStream is = new ByteArrayInputStream(body.getBytes("UTF-8"));
+                final Model model = ModelFactory.createDefaultModel();
+
+                model.read(is, location, "JSON-LD");
+                if (model.contains(locResource, RdfType)) {
```

Comment:
You're not checking the agent value here. Should you be?

Reply:
Should you check that each event has its own unique identifier? (This is different than the object IRI)

Reply:
This test is all the `MUST` requirements so as Agent is a `SHOULD` I was putting those tests below in 6.2-B. As for unique identifier for each event,  I don't think there is that requirement in the specification so while it makes sense I don't know if we need to test for it.